File: backend/src/main/java/com/devops/dashboard/service/AuthService.java
package com.devops.dashboard.service;

import com.devops.dashboard.dto.auth.AuthResponse;
import com.devops.dashboard.dto.auth.LoginRequest;
import com.devops.dashboard.dto.auth.RegisterRequest;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
}

File: frontend/src/services/authService.js
import { api } from './api';

export const authService = {
  login: async (payload) => {
    const response = await api.post('/auth/login', payload);
    return response.data;
  },
  register: async (payload) => {
    const response = await api.post('/auth/register', payload);
    return response.data;
  },
};
